metadataStore/odm_templates.py

```diff
@@ -113,24 +146,23 @@ class Event(Document):
 
     Attributes
     ----------
-    descriptor : metadataStore.odm_templates.EventDescriptor
-        Foreign key to EventDescriptor that provides info regarding a
-        set of events of the same type
-
-    data : dict
-        Data dictionary where experimental data is stored.
-
-    time : timestamp
-        The date/time as found at the client side when an event is
-        created.
+    descriptor : mongoengine.DynamicDocument
+        EventDescriptor foreignkey
+    uid : str
+        Globally unique identifier for this Event
+    seq_num: int
+        Sequence number to identify the location of this Event in the Event stream
+    data: dict
+        Dict of lists that contain e.g. data = {'motor1': [value, timestamp]}
+    time : float
+        The event time.  This maybe different than the timestamps on each of the data entries
 
-    seq_no : int, optional
-        Sequence number pointing out the
     """
     descriptor = ReferenceField(EventDescriptor,reverse_delete_rule=DENY,
                                 required=True, db_field='descriptor_id')
-    seq_no = IntField(min_value=0, required=True)
+    uid = StringField(required=True, unique=True)
+    seq_num = IntField(min_value=0, required=True)
     data = DictField(required=True)
     time = FloatField(required=True)
     time_as_datetime = DateTimeField(required=False)
-    meta = {'indexes': ['-descriptor', '-_id', '-time']}
+    meta = {'indexes': ['-descriptor', '-_id']}
```

Comment:
That makes sense. the way to check if it is unqiue is to make sure a search fails before insertion and if you are going to search on every insert, you need to index it.